.github/workflows/test-in-app-messaging-prs.yml

```diff
@@ -0,0 +1,63 @@
+# Description: This workflow runs unit + e2e tests on PRs targeting
+#   - `in-app-messaging/main`
+#   - `in-app-messaging/release`
+#
+# Triggered by:
+#   (1) Internal PRs: maintainer pushes a commit to PRs targeting those branches.
+#   (2) Fork PRs: maintainer adds "run-test" label to PRs targeting those branches.
+
+name: Test / in-app-messaging / PRs
+
+concurrency:
+  group: e2e-${{ github.event.pull_request.id }}
+  cancel-in-progress: true
+
+on:
+  pull_request_target:
+    branches: [
+        in-app-messaging/main
+        in-app-messaging/release,
+      ]
+    types: [opened, synchronize, labeled]
+
+permissions:
+  pull-requests: write # used to remove label
+  # other permissions are defaulted to "none"
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    # We run tests only if it's:
+    #   1) pull request not from a fork (ie. internal PR), or
+    #   2) pull request from a fork (ie. external PR) that was added "run-tests" label
+    if: |
+      (github.event.pull_request.head.repo.full_name == github.repository) ||
```

Comment:
nit: is there a way to put this in a function? I think we use this conditional in multiple workflows.